lib/dfe/analytics/event.rb

```diff
@@ -78,6 +79,12 @@ def with_data(hash)
         self
       end
 
+      def with_hidden_data(hash)
+        @event_hash.deep_merge!(hidden_data: hash_to_kv_pairs(hash))
+
```

Comment:
This possibly doesn't need to be deep_merge here if the hidden_data we are sending is not going to be nested

Reply:
This is OK I believe, as we can't make assumptions about nesting within the hidden data.